pkg/geoip/geolite2.go

```diff
@@ -20,9 +30,80 @@ type geoLite2 struct {
 	db *geoip2.Reader
 }
 
+// downloadGeoLite2Db downloads the GeoLite2 database and extract the files into the DBPath.
+// The files extracted are: COPYRIGHT.txt, LICENSE.txt and GeoLite2-City.mmdb.
+func downloadGeoLite2Db(maxmindDBLicense string) error {
+	// it makes a http request to download the GeoLite2Db and return the bytes to a .tar.gz file with the database inside it.
+	r, err := http.Get(fmt.Sprintf("https://download.maxmind.com/app/geoip_download?edition_id=GeoLite2-City&license_key=%s&suffix=tar.gz", maxmindDBLicense))
+	if err != nil {
+		return err
+	}
+
+	extractGeoLite2Db := func(geoLite2Db io.Reader, directory string) error {
```

Comment:
Please use the `Untar` method from https://github.com/fluxcd/pkg/tree/untar/v0.1.0